bittensor/commands/profile.py

```diff
@@ -0,0 +1,528 @@
+                sub_content[k] = {}
+            sub_content = sub_content[k]
+        updated = keys[-1] in sub_content
+        sub_content[keys[-1]] = value
+        return updated
+
+    @staticmethod
+    def check_config(config: "bittensor.config"):
+        return config is not None
+
+    @staticmethod
+    def add_args(parser: argparse.ArgumentParser):
+        profile_parser = parser.add_parser("set_value", help="""Set or update profile values""")
+        profile_parser.set_defaults(func=ProfileSetValueCommand.run)
+        profile_parser.add_argument("--profile.name", type=str, help="The name of the profile to update")
+        profile_parser.add_argument("--profile.path", type=str, default=defaults.profile.path,
+                                    help="The path to the profile directory")
+        profile_parser.add_argument("args", nargs=argparse.REMAINDER, help="The key-value pairs to set or update")
+
+
+class ProfileDeleteValueCommand:
+    @staticmethod
+    def run(cli):
+        ProfileDeleteValueCommand._run(cli)
+
+    @staticmethod
+    def _run(cli: "bittensor.cli"):
+        config, profile_path, contents = open_profile(cli, "delete_value")
+
+        if profile_path is None:
+            return
+
+        for arg in cli.config.args:
+            key = arg
+            if ProfileDeleteValueCommand._remove_key(contents, key):
+                log_info(f"Variable {key} was removed from profile {config.profile.name}")
+            else:
+                log_warning(f"Variable {key} does not exist in profile {config.profile.name}")
+
+        try:
+            with open(profile_path, "w") as f:
+                yaml.safe_dump(contents, f)
+        except Exception as e:
+            handle_error("Failed to write profile", e)
+
+    @staticmethod
+    def _remove_key(contents, key):
+        keys = key.split(".")
+        sub_content = contents
+        for k in keys[:-1]:
+            sub_content = sub_content.get(k, {})
+        return sub_content.pop(keys[-1], None) is not None
+
+    @staticmethod
+    def check_config(config: "bittensor.config"):
+        return config is not None
+
+    @staticmethod
+    def add_args(parser: argparse.ArgumentParser):
+        profile_parser = parser.add_parser("delete_value", help="""Delete profile values""")
+        profile_parser.set_defaults(func=ProfileDeleteValueCommand.run)
+        profile_parser.add_argument("--profile.name", type=str, help="The name of the profile to update")
+        profile_parser.add_argument("--profile.path", type=str, default=defaults.profile.path,
+                                    help="The path to the profile directory")
+        profile_parser.add_argument("args", nargs=argparse.REMAINDER, help="The keys to delete")
+
+
+class ProfileSetCommand:
+    @staticmethod
+    def run(cli):
+        config, profile_path, contents = open_profile(cli, "set", True)
+
+        # Load generic config file and write active profile to it
```

Comment:
I don't think that's what this is supposed to do. Changing the active profile just changes what profile values are used if the --profile.name parameter isn't specified at the cli. Please review what I had done in my code change and see if you can reuse that here.